tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -197,6 +197,41 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "support-list",
 	},
 
+	/** Firmware layout for the CPE210 V3 */
+	{
+		.id     = "CPE210V3",
+		.vendor = "CPE210(TP-LINK|UN|N300-2|00000000):3.0\r\n",
+		.support_list =
+			"SupportList:\r\n"
+			"CPE210(TP-LINK|EU|N300-2|45550000):3.0\r\n"
+			"CPE210(TP-LINK|UN|N300-2|00000000):3.0\r\n"
+			"CPE210(TP-LINK|UN|N300-2):3.0\r\n"
+			"CPE210(TP-LINK|EU|N300-2):3.0\r\n",
+		.support_trail = '\xff',
+		.soft_ver = NULL,
+
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x20000},
+			{"partition-table", 0x20000, 0x01000},
+			{"default-mac", 0x30000, 0x00020},
+			{"product-info", 0x31100, 0x00100},
+			{"device-info", 0x31400, 0x00400},
+			{"signature", 0x32000, 0x00400},
+			{"device-id", 0x33000, 0x00100},
+			{"firmware", 0x40000, 0x7B0000},
```

Comment:
This should be:
{"firmware", 0x40000, 0x770000} ?

Reply:
yup

Reply:
i also will try to build the current version with all the latest commits i have done on my CPE210v3

Reply:
Hi,
was trying to compile the version - but it ends with a core dump during build of the sysupgrade bin image. 
I had to change in tplink-safeloader.c the ".first_sysupgrade_partition" from firmware to os-partition. 

Did you have the same problem ? 
I saw this is set even on other devices, even they don't have a partition with this name. 
(e.g. ARCHER-A7-V5)

Reply:
I did not try yet, but I think you are right. However, it is not "os-partition", but "os-image". Looks like this is already anticipating the splitting of "firmware" into "os-image" and "file-system".
I also do not really understand why sometimes the last partition is "support-list" and sometimes it is "file-system". However, I would stick to the other CPExxx which use "support-list" in this case.
If it works, you should again squash your commits into one ...